src/vxingest/builder_common/vx_ingest.py

```diff
@@ -186,12 +187,19 @@ def get_file_list(self, df_query, directory, file_pattern):
             result = self.cluster.query(df_query)
             df_elements = list(result)
             df_full_names = [element["url"] for element in df_elements]
-            if Path(directory).exists() and Path(directory).is_dir():
+            if pathlib.Path(directory).exists() and pathlib.Path(directory).is_dir():
                 file_list = sorted(
-                    Path(directory).glob(file_pattern), key=os.path.getmtime
+                    pathlib.Path(directory).glob(file_pattern), key=os.path.getmtime
                 )
                 for filename in file_list:
                     try:
+                        # test to see if this filename can be parsed into a valid date using the fmask, i.e. it is a valid file name
+                        try:
+                            _d = dt.datetime.strptime(
+                                pathlib.PurePath(filename).name, file_mask
+                            )
+                        except ValueError:
+                            continue
```

Comment:
Is the purpose of this chunk solely to filter out dot files? If so, it might be more straightforward to do so directly instead of trying to match a datetime mask. 

E.g.
```python
def is_regular_path(path: Path) -> bool:
    """Filter function to detect if there is a dotfile or hidden directory in a Path object"""
    parts = path.parts
    return not any(part.startswith('.') for part in parts)

def get_file_list(self, df_query, directory, file_pattern):
  # ...
  file_list = sorted(Path(directory).glob(file_pattern), key=os.path.getmtime)
  # Remove any Paths with dotfiles or hidden dirs from our list
  file_list = filter(is_regular_path, file_list)
  for filename in file_list:
    # etc...
```

Otherwise - it looks like the `file_mask` is supposed to be the same as the `file_pattern` which seems confusing to me. I also don't think using `pathlib.Path.name` will filter out dot directories like we're hoping. That method returns the file name so `.prev/202109...` and `validDir/202109...` will both return `202109...` and pass the test.

Reply:
We could then test the filter function with some pretty trivial parameterized unit tests like so:

```python
@pytest.mark.parameterize(
  "path,expected",
  [
    (Path(tmp_path / "1820013010000"), False),
    (Path(tmp_path / ".prev" / "1820013010000"), False),
    (Path(tmp_path / ".tmp" / "1820013010000"), True),
    # etc...
  ]
def test_is_regular_path(path, expected):
  assert is_regular_path(path) == expected
```